TicTacToe2/TicTacToe/Board.cs

```diff
@@ -70,6 +70,45 @@ public IEnumerable<char[]> GetDimensions()
 
             // fill dimensions with data from spaces
 
+            // horizontal
+            char[] x = new char[Size];
+            for (int row = 0; row < Size; row++)
+            {
+
+                for (int col = 0; col < Size; col++)
+                {
+                    x[row] = spaces[row, col];
+                }
+
+                dimensions.Add(x);
+            }
+
+            // vertical
+            for (int col = 0; col < Size; col++)
+            {
+
+                for (int row = 0; row < Size; row++)
+                {
+                    x[col] = spaces[row, col];
+                }
+
+                dimensions.Add(x);
+            }
+
+            // diagonal
+            for (int row = 0; row < Size; row++)
+            {
+                x[row] = spaces[row, row];
```

Comment:
There are some issues with the logic for diagonals. Consider a 3x3 grid. It should have a total of 8 dimensions:

* 3 rows
* 3 columns
* 1 diagonal up ⬈
* 1 diagonal down ⬊

How many arrays is this loop adding to the dimensions list?